pkg/pipeline/ingest/ingest_collector.go

```diff
@@ -180,7 +184,14 @@ func (ingestC *ingestCollector) processLogLines(out chan<- []interface{}) {
 		case <-flushRecords.C:
 			// Process batch of records (if not empty)
 			if len(records) > 0 {
-				log.Debugf("ingestCollector sending %d entries", len(records))
+				if len(ingestC.in) > 0 {
```

Comment:
fix is here, if the out channel is full, then we only loop once or twice in the ingest case before looping again in this case.